akka-projection-core/src/main/scala/akka/projection/internal/SourceProviderAdapter.scala

```diff
@@ -48,7 +55,9 @@ import akka.stream.javadsl.{ Source => JSource }
 @InternalApi private[projection] class ScalaBySlicesSourceProviderAdapter[Offset, Envelope](
```

Comment:
We have two of these adapters, one here in core and then another in projection-r2dbc. Looks very similar, if not identical?

Reply:
Wonder if the reason we had two was that we didn't want to make the assumption that EventTimestampQuery and LoadEventQuery are always implemented. For r2dbc we know they are.

Reply:
I don't remember, but gut feeling is that this could be simplified. Like one implementation here with an apply that chooses what mixins the adapter should have.

Reply:
I'll give it a shot pulling those into core, adding factories there and giving them more clear names.

Reply:
Moved/refactored/renamed in [90430a6](https://github.com/akka/akka-projection/pull/1094/commits/90430a65825baaed26e4a19d994185db6c5171bd)